src/sensor/ping.cpp

```diff
@@ -120,6 +99,34 @@ Ping::Ping() : Sensor()
     detectorThread()->start();
 }
 
+void Ping::startPreConfigurationProcess()
+{
+    qCDebug(PING_PROTOCOL_PING) << "Start pre configuration task and requests.";
+    if(!link()->isWritable()) {
+        qCDebug(PING_PROTOCOL_PING) << "It's only possible to set last configuration when link is writable.";
+        return;
+    }
+
+    setAutoDetect(false);
+    SettingsManager::self()->lastLinkConfiguration(*link()->configuration());
+
+    // Request device information
+    request(Ping1DNamespace::Mode_auto);
+    request(Ping1DNamespace::Profile);
+    request(Ping1DNamespace::Firmware_version);
+    request(Ping1DNamespace::Device_id);
+    request(Ping1DNamespace::Speed_of_sound);
+
+    // Start periodic request timer
+    _periodicRequestTimer.start();
+
+    // Save configuration
+    SettingsManager::self()->lastLinkConfiguration(*link()->configuration());
+
+    // Start configuration
+    setLastPingConfiguration();
```

Comment:
Are we guaranteeded that we will have the correct srcId for the device at this point so that we load the correct settings?